resources/other/git.md

```diff
@@ -1,46 +1,171 @@
+
+# TOC
+
+* Recipies
+
+ 1. [Getting a repo](#clone)
+ 1. [Receiving work](#pull)
+ 1. [Adding a file](#add)
+ 1. [Committing work](#commit)
+ 1. [Sharing work](#push)
+ 1. [Listing existing branches](#list)
+ 1. [Switching to a branch](#switch)
+ 1. [Moving work into a new branch](#move) 
+
+* [Additional content](#additional)
+
+* [Resources](#resources)
+
+<a name="clone"></a>
+## Getting a repo 
+
+Before you can use a repo you need to get a local copy on your machine from the remote server. To do this, you can specify the repo you want checked out.
+
+`git clone git@github.students.cs.ubc.ca:<org>/<repo>.git`
+
+This should work for public repos, but if you are checking out a private repo you may need to configure your username and password.
+
+<a name="pull"></a>
+## Receiving work
+
+If there have been updates on your repo (e.g., your teammates have made commits), you can update your local copy with any changes they have shared with the remote server.
+
+`git pull` 
+
+<a name="add"></a>
+## Adding a file
+
+Files you create locally on disk need to be explicitly added to Git so they can be tracked. To add a file:
+
+`git add <fileName>`
+
+If you are wondering what files are untracked:
+
+`git status .`
+
+<a name="commit"></a>
+## Committing work
+
+After completing work, you need to complete it so the current status of your tracked files can be saved in the repo. To commit the changes you have made to all files at once:
+
+`git commit -a -m "<message>"`
+
+If you just want to commit changes to a single file:
+
+`git commit -m "<message>" <fileName>`
+
+<a name="push"></a>
+## Sharing work
+
+So far your commits just exist on your local machine. To share them with others, you will need to upload them to the server:
+
+`git push`
+
+Note: if the server version has been updated since you last received work, you will have to perform a `git pull` before the `git push` will work.
+
+<a name="list"></a>
+## Listing existing branch names
+
+Much work in Git takes place on branches. Branches are the right place for _in progress_ development work. When using branches, the main branch is reserved for _completed_ work. This allows multiple people to work independently without overwriting each other's work, although waiting too long between merging a development branch with main increases the chances of merge conflicts. To get a list of branches you can switch to, run:
+
+`git branch -r`
+
+<a name="switch"></a>
+## Switching to a branch
+
+If you are in a repo that has not had any changes to it (e.g., no file changes and no commits), you can change cleanly to another branch. If `branchName` exists, you will switch to the branch. If it does not exist, it will be created. Git branch names cannot contain spaces.
```

Comment:
If you want to change to a new branch I think you need the `-b` flag?

Reply:
I checked this (and tested it). -b just creates the branch if it doesn't exist (went with this case after list since the next case uses -b to create a new branch)